api/app/tests/medium/routers/test_actionlogs.py

```diff
@@ -106,73 +174,176 @@ def test_create_log__with_wrong_params(testdb: Session):
+        create_actionlog_with_related_data(
+            user1.user_id,
             USER1,
+            topic1,
             action1.action_id,
-            topic1.topic_id,
-            user1.user_id,
             uuid4(),
             service1.service_id,
+            tag_id,
+            dependency1_id,
+            threat1_id,
+            ticket1_id,
             None,
+            testdb,
         )
+
+
+def test_create_log__with_called_by_not_pteam_member(testdb: Session):
+    user1 = create_user(USER1)
+    create_user(USER2)
+    pteam1 = create_pteam(USER1, PTEAM1)
+    upload_pteam_tags(USER1, pteam1.pteam_id, SERVICE1, {TAG1: [("Pipfile.lock", "1.0.0")]}, True)
+    pteam1_model = persistence.get_pteam_by_id(testdb, pteam1.pteam_id)
+    assert pteam1_model is not None
+    service1 = pteam1_model.services[0]
+    tag_id = service1.dependencies[0].tag.tag_id
+    topic1 = create_topic(USER1, TOPIC1, actions=[ACTION1])
+    action1 = topic1.actions[0]
+
+    dependency1_id = "dependency1_id"
+    threat1_id = "threat1_id"
+    ticket1_id = "3d362f0f-e08e-45a3-9ae9-5a46936372c0"
+
     # called by not pteam member
     with pytest.raises(HTTPError, match="403: Forbidden"):
-        create_actionlog(
+        create_actionlog_with_related_data(
+            user1.user_id,
             USER2,
+            topic1,
             action1.action_id,
-            topic1.topic_id,
-            user1.user_id,
             pteam1.pteam_id,
             service1.service_id,
+            tag_id,
+            dependency1_id,
+            threat1_id,
+            ticket1_id,
             None,
+            testdb,
         )
+
+
+def test_create_log__with_not_pteam_member_as_recipient(testdb: Session):
+    create_user(USER1)
+    user2 = create_user(USER2)
+    pteam1 = create_pteam(USER1, PTEAM1)
+    upload_pteam_tags(USER1, pteam1.pteam_id, SERVICE1, {TAG1: [("Pipfile.lock", "1.0.0")]}, True)
+    pteam1_model = persistence.get_pteam_by_id(testdb, pteam1.pteam_id)
+    assert pteam1_model is not None
+    service1 = pteam1_model.services[0]
+    tag_id = service1.dependencies[0].tag.tag_id
+    topic1 = create_topic(USER1, TOPIC1, actions=[ACTION1])
+    action1 = topic1.actions[0]
+
+    dependency1_id = "dependency1_id"
+    threat1_id = "threat1_id"
+    ticket1_id = "3d362f0f-e08e-45a3-9ae9-5a46936372c0"
+
     # not pteam member as recipient
     with pytest.raises(HTTPError, match="400: Bad Request"):
-        create_actionlog(
+        create_actionlog_with_related_data(
+            user2.user_id,
             USER1,
+            topic1,
             action1.action_id,
-            topic1.topic_id,
-            user2.user_id,
             pteam1.pteam_id,
             service1.service_id,
+            tag_id,
+            dependency1_id,
+            threat1_id,
+            ticket1_id,
             None,
+            testdb,
         )
+
+
+def test_create_log__with_action_mismatch_with_topic(testdb: Session):
```

Comment:
```suggestion
def test_it_should_return_400_when_create_log_with_action_mismatch_with_topic(testdb: Session):
```
`mismatch` とは何を指してるかがわからないので、正確に記述してください

`create_actionlog_with_related_data` でデータやactionlog リクエストが隠蔽されてて何を検証したいかがコードから読み取れないので、そこがわかるようにして欲しいです (actionlogs へのpostリクエストは隠蔽されないでほしい）

Reply:
TopicActionが持つtopic_idが指定したtopic_idと一致しないケースのため、テスト名を下記に変更しました。
test_it_should_return_400_when_create_log_with_action_not_belong_specified_topic

その他のテスト名は上記指摘の通り修正しました。

ActionLog生成前にTicketを作るデータ準備について、直接DBにTicketレコードをINSERTする方法から、
upload_pteam_tagsで作ったTagに対応するTopicを生成することで内部的にTicketが生成される方式に変更しました。
これにより、actionlog リクエストのデータ隠蔽が軽減されました。

また、この変更に伴い、下記テストがNGとなっていた問題が解消しました。
・test_actionlogs.py::test_get_logs__members_only()